straxen/corrections_services.py

```diff
@@ -350,20 +371,41 @@ def apply_cmt_version(context: strax.Context, cmt_global_version: str):
     # we want this error to occur in order to keep fixed global versions
     cmt_config = dict()
     failed_keys = []
-    for option, tup in cmt_options.items():
-        try:
-            # might need to modify correction name to include position reconstruction algo
-            correction_name = tup[0]
+
+    for option, value in cmt_options.items():
+        # might need to modify correction name to include position reconstruction algo
+        # if the config is a URL, get the correction name in CMT
+        if isinstance(value, str) and 'cmt://' in value:
+            correction_name = correction_name_from_urlstring(value)
+        # if it's still a tuple, it's the first element
+        else:
+            correction_name = value[0]
+            # this is a bit of a mess, but posrec configs are treated differently in the tuples
             if correction_name in posrec_corrections_basenames:
                 correction_name += f"_{posrec_algo}"
-            new_tup = (tup[0], local_versions[correction_name], tup[2])
-        except KeyError:
-            failed_keys.append(option)
+
+        # now see if our correction is in our local_versions dict
+        if correction_name in local_versions:
+            if isinstance(value, str) and 'cmt://' in value:
+                new_value = replace_url_version(value, local_versions[correction_name])
+            # if it is a tuple, make a new tuple
+            else:
+                new_value = (value[0], local_versions[correction_name], value[2])
+        else:
+            failed_keys.append(correction_name)
             continue
-        cmt_config[option] = new_tup
+
+        cmt_config[option] = new_value
+
     if len(failed_keys):
         failed_keys = ', '.join(failed_keys)
-        raise CMTVersionError(f"CMT version {cmt_global_version} is not compatible with this straxen version! "
-                              f"CMT {cmt_global_version} is missing these corrections: {failed_keys}")
+        msg = f"CMT version {cmt_global_version} is not compatible with this straxen version! " \
+              f"CMT {cmt_global_version} is missing these corrections: {failed_keys}"
+
+        # only raise a warning if we are working with the online context
+        if cmt_global_version == "global_ONLINE":
+            warnings.warn(msg)
+        else:
+            raise CMTVersionError(msg)
```

Comment:
Shouldn't we just quit if we are using the online version? I don't see why this function would be called at all of online?

Reply:
Yeah I think it wouldn't be called typically, and that is also why I changed to warning instead of raising an error. So you're saying don't even bother with the warning?

Reply:
yeah, it's hard to figure out what be the correct way of dealing with abuse (what I would call throwing an ONLINE version into this function) of this code would be.

I would indeed just raise the error (maybe _especially_) if someone is using it on an ONLINE version.

Reply:
I don't think we want to raise an error for online, as if you hit this problem it just means a new correction has not been added to the global_online collection yet (so defaulting to default values, which should use online _local_ correction anyway, is not an issue). It's the fixed offline versions that need an error as those should never change so will never had the new corrections we add. 

Personally I think a warning seems like good middle ground. It notifies you that something is maybe amiss but doesn't it's not a huge deal. Plus it was nice to test this function with global_ONLINE yesterday since the new offline is not out yet :)

Reply:
ah okay, you just want to have it as a check of the `global` collection of CMT. Okay, this I would not have guessed 😉 because it looks quite funny if one does:
 - start always from `xenonnt_online` (with online local versions) 
 - for `xenonnt_vX` we do `xenonnt_vX = xenonnt_online().apply_cmt_version("global_vX")` 
 - but then you want to do: `xenonnt_vX().apply_cmt_version("global_ONLNE")`.

Reply:
It does look quite funny, I agree. But it's not wrong in principle right?